bug: temporary value dropped while borrowed

### Describe the bug

On Rust 1.76 (rustic MSRV) we have a build failure due to:

```
   error[E0716]: temporary value dropped while borrowed
    --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/opendal-0.49.1/src/services/fs/lister.rs:90:14
     |
  88 |         let p = if metadata.is_dir() {
     |             - borrow later stored here
  89 |             // Make sure we are returning the correct path.
  90 |             &format!("{rel_path}/")
     |              ^^^^^^^^^^^^^^^^^^^^^^ creates a temporary value which is freed while still in use
  91 |         } else {
     |         - temporary value is freed at the end of this statement
     |
     = note: consider using a `let` binding to create a longer lived value
     = note: this error originates in the macro `format` (in Nightly builds, run with -Z macro-backtrace for more info)
```

I assume this is caught during us checking against opendal on 1.76 while your MSRV is 1.75.

### Steps to Reproduce

`rustup run 1.76 cargo check` should spit the above error

### Expected Behavior

It should build.

### Additional Context

Source in CI logs: https://github.com/rustic-rs/rustic_core/actions/runs/10576555813/job/29302600762#step:4:606

### Are you willing to submit a PR to fix this bug?

- [X] Yes, I would like to submit a PR.

--- a/core/src/services/fs/lister.rs
+++ b/core/src/services/fs/lister.rs
@@ -152,13 +152,13 @@ impl oio::BlockingList for FsLister<std::fs::ReadDir> {
             meta
         };
 
-        let p = if metadata.is_dir() {
+        let entry = if metadata.is_dir() {
             // Make sure we are returning the correct path.
-            &format!("{rel_path}/")
+            oio::Entry::new(&format!("{rel_path}/"), metadata)
         } else {
-            &rel_path
+            oio::Entry::new(&rel_path, metadata)
         };
 
-        Ok(Some(oio::Entry::new(p, metadata)))
+        Ok(Some(entry))
     }
 }

